pkg/chunk/composite_store.go

```diff
@@ -101,28 +101,42 @@ func (c compositeStore) Get(ctx context.Context, userID string, from, through mo
 // LabelValuesForMetricName retrieves all label values for a single label name and metric name.
 func (c compositeStore) LabelValuesForMetricName(ctx context.Context, userID string, from, through model.Time, metricName string, labelName string) ([]string, error) {
 	var result []string
+	uniqueValues := map[string]struct{}{}
 	err := c.forStores(from, through, func(from, through model.Time, store Store) error {
 		labelValues, err := store.LabelValuesForMetricName(ctx, userID, from, through, metricName, labelName)
 		if err != nil {
 			return err
 		}
-		result = append(result, labelValues...)
+		for _, l := range labelValues {
```

Comment:
Since this piece of code is getting duplicated in few places, have you considered creating an utility function `appendUniqueStrings(result []string, uniqueValues map[string)struct{}, values []string) []string`?

I would be easily called like this:
```
result = appendUniqueStrings(result, uniqueVaLues, labelValues)
```